scrapegraphai/graphs/abstract_graph.py

```diff
@@ -221,15 +204,18 @@ def handle_model(model_name, provider, token_key, default_token=8192):
                 raise KeyError("Model not supported") from exc
             return OneApi(llm_params)
 
-        if "nvidia" in llm_params["model"]:
+        elif "nvidia" in llm_params["model"]:
+
             try:
                 self.model_token = models_tokens["nvidia"][llm_params["model"].split("/")[-1]]
                 llm_params["model"] = "/".join(llm_params["model"].split("/")[1:])
             except KeyError as exc:
                 raise KeyError("Model not supported") from exc
             return ChatNVIDIA(llm_params)
+        else:
```

Comment:
`handle_model` may raise unhandled errors if provided with unsupported models and providers. It should be enclosed in a `try/except` block, and / or a specific condition (`if model_name...` with the remaining four providers) should be added